tensorflow/python/keras/backend.py

```diff
@@ -5530,47 +5530,24 @@ def bias_add(x, bias, data_format=None):
     raise ValueError(
         'Unexpected bias dimensions %d, expect to be 1 or %d dimensions' %
         (len(bias_shape), ndim(x)))
-  # pylint: disable=g-no-augmented-assignment
+
+  if len(bias_shape) == 1:
+    if data_format == 'channels_first':
+      return nn.bias_add(x, bias, data_format='NCHW')
+    return nn.bias_add(x, bias, data_format='NHWC')
   if ndim(x) == 5:
     if data_format == 'channels_first':
-      if len(bias_shape) == 1:
-        x = x + reshape(bias, (1, bias_shape[0], 1, 1, 1))
-      else:
-        x = x + reshape(bias, (1, bias_shape[3]) + bias_shape[:3])
-    elif data_format == 'channels_last':
-      if len(bias_shape) == 1:
-        x = x + reshape(bias, (1, 1, 1, bias_shape[0]))
-      else:
-        x = x + reshape(bias, (1,) + bias_shape)
-  elif ndim(x) == 4:
+      return x + reshape(bias, (1, bias_shape[3]) + bias_shape[:3])
```

Comment:
I think this section can be further reduced.

```

if ndim(x) in (3, 4, 5):
  if data_format = 'channels_first':
     bias_reshape_axis = (1, bias_shape[-1]) + bias_shape[:-1]
     return x + reshape(bias, bias_reshape_axis)
  return x + reshape(bias, (1, ) + bias_shape)
```

Reply:
@qlzh727 Great idea, fixed it in the latest commit.